### PR TITLE
Acom 771 quizzes

### PR DESCRIPTION
**Pull request by jaredplanter**
This adds support for Quizzes in ACTV, the new objects are: Quiz, QuizOutcome, QuizQuestion and QuizAnswer.

### Testing
To test update your 'actv' entry in your A3 GemFile to ->
 `gem 'actv', github: 'activenetwork/actv', branch: 'acom-771-quizzes'`
Now open `rails console` and follow the instructions below.
```ruby
results = ACTV.search(nil, categoryName: 'quiz').results

#Quiz - outputs a single ACTV::Quiz object
quiz = results.first
#Questions - outputs an array of ACTV::QuizQuestion objects
questions = quiz.questions

#Example - drilling down into an outcome
question = questions.first #ACTV::QuizQuestion object
answers = question.answers #Array of ACTV::QuizAnswer objects
answer = answers.first #ACTV::QuizAnswer object
outcome = answer.outcome #ACTV::QuizOutcome object (returns nil temporarily - read below)
```

The `outcome` variable will return `nil` right now because we are in the process of adding an assetReferenceType to the asset service database, without a registered type we cannot add references. Please pull the branch down and run the specs to very that code is working properly.

### CODE REVIEW

File: lib/actv/asset.rb [hunk 1]
```diff
@@ -373,8 +373,20 @@ def references
...
+      end
+    end
+
     private
 
+    def children_assets_filtered_by_category category
+      components.select do |component|
+        component.full_asset.category_is? category
+      end.compact.map &:full_asset
+    end
```
Comment (hlcfan) [inline lib/actv/asset.rb]: What do you think do it in a bulk query like
``` ruby
cmp_ids = components.map &:assetGuid
cmps = ACTV.asset cmp_ids
cmps.select { |c| c.category_is? category }
```

File: lib/actv/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module ACTV
-  VERSION = "2.0.0"
+  VERSION = "2.1.0"
```
Comment (mattr-) [inline lib/actv/version.rb]: Can we take the version bump out of this PR? With the upcoming video support for ED pages, I think it makes more sense to bump the version separately from both of these PRs.

File: spec/actv/asset_component_spec.rb [hunk 3]
```diff
@@ -8,7 +8,6 @@
   describe "#prices" do
     before(:each) do
       stub_request(:post, "http://api.amp.active.com/v2/assets.json").
-        with(:body => {"id"=>""}, :headers => {'Accept'=>'application/json', 'Content-Type'=>'application/x-www-form-urlencoded', 'User-Agent' => "Active Ruby Gem #{ACTV::VERSION}"}).
```
Comment (brdrgz) [inline spec/actv/asset_component_spec.rb]: shouldn't need to remove this since it's fixed in master now, but I guess we can anyway

**Comment by brdrgz** [comment]
looks good to me :ok_hand:

**Pull request merged**